sourcegraph/client.go

```diff
@@ -302,22 +302,20 @@ func addOptions(u *url.URL, opt interface{}) error {
 	return nil
 }
 
-// keepAliveTransport returns a http.RoundTripper that uses a larger
+// keepAliveTransport is an http.RoundTripper that uses a larger
 // keep-alive pool than the default.
-func keepAliveTransport() http.RoundTripper {
-	return &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		Dial: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-		}).Dial,
-		TLSHandshakeTimeout: 10 * time.Second,
-
-		// Allow more keep-alive connections per host to avoid
-		// ephemeral port exhaustion due to getting stuck in
-		// TIME_WAIT. Some systems have a very limited ephemeral port
-		// supply (~1024). 20 connections is perfectly reasonable,
-		// since this client will only ever hit one host.
-		MaxIdleConnsPerHost: 20,
-	}
+var keepAliveTransport http.RoundTripper = &http.Transport{
```

Comment:
Weird, so it looks like Transport doesn't close connections as long as the number of connections is below `MaxIdleConnsPerHost`... even if the Transport becomes unreference-able and is garbage collected. Is there a destructor-like pattern in Go for this scenario?